components/micro-gateway-core/src/main/java/org/wso2/micro/gateway/core/mutualssl/CertificateUtils.java

```diff
@@ -34,19 +37,24 @@
  * This class is responsible for do certificate level functionalities.
  */
 public class CertificateUtils {
+    private static final Logger log = LogManager.getLogger(CertificateUtils.class);
 
-    public static String getAliasFromHeaderCert(String base64EncodedCertificate) throws  KeyStoreException,
-            java.security.cert.CertificateException, CertificateException {
-        base64EncodedCertificate = URLDecoder.decode(base64EncodedCertificate).
-                replaceAll(Constants.BEGIN_CERTIFICATE_STRING, "")
-                .replaceAll(Constants.END_CERTIFICATE_STRING, "");
-        byte[] bytes = Base64.decodeBase64(base64EncodedCertificate);
-        InputStream inputStream = new ByteArrayInputStream(bytes);
-        X509Certificate x509Certificate = X509Certificate.getInstance(inputStream);
-        if (getAliasFromTrustStore(x509Certificate, LoadKeyStore.trustStore) != null) {
-            return getAliasFromTrustStore(x509Certificate, LoadKeyStore.trustStore);
-        } else {
-            return "";
+    public static String getAliasFromHeaderCert(String base64EncodedCertificate) {
+        try {
+            base64EncodedCertificate = URLDecoder.decode(base64EncodedCertificate).
+                    replaceAll(Constants.BEGIN_CERTIFICATE_STRING, "").replaceAll(Constants.END_CERTIFICATE_STRING, "");
+            byte[] bytes = Base64.decodeBase64(base64EncodedCertificate);
+            InputStream inputStream = new ByteArrayInputStream(bytes);
+            X509Certificate x509Certificate = X509Certificate.getInstance(inputStream);
+            if (getAliasFromTrustStore(x509Certificate, LoadKeyStore.trustStore) != null) {
+                return getAliasFromTrustStore(x509Certificate, LoadKeyStore.trustStore);
+            } else {
+                return "";
+            }
+        } catch (KeyStoreException | java.security.cert.CertificateException | CertificateException e) {
+            String msg = "Error while decoding certificate present in the header and validating with the trust store.";
+            log.error(msg, e);
+            throw ErrorUtils.getBallerinaError(msg, e);
```

Comment:
If when throwing the error here will also log the error msg, then calling log.error() again may not be needed.

Reply:
If we do not log error here, it would not log the stack trace properly. We need to log the error here properly, and pass the error to ballerina level to handle from there.